pkg/controller/migcluster/validation.go

```diff
@@ -216,6 +226,47 @@ func (r ReconcileMigCluster) testConnection(cluster *migapi.MigCluster) error {
 	return nil
 }
 
+// Validate the Exposed registry route
+func (r ReconcileMigCluster) validateRegistryRoute(cluster *migapi.MigCluster) error {
+
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	if cluster.Spec.ExposedRegistryPath != "" {
+		url := "https://" + cluster.Spec.ExposedRegistryPath + "/v2/"
+		restConfig, err := cluster.BuildRestConfig(r.Client)
+		token := restConfig.BearerToken
+
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		client := &http.Client{Transport: tr}
+
+		req, err := http.NewRequest("GET", url, nil)
+
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+
+		req.Header.Set("Authorization", "bearer "+token)
+
+		res, err := client.Do(req)
+		if err != nil {
```

Comment:
This should not only check for error, but it should also be performing a check on the status code to ensure it returns a 200.

This is failing for me if I do a valid URL but an invalid route to a registry.

Reply:
Thanks for the catch. Updated the PR.